src/Functions/base58Encode.cpp

```diff
@@ -15,7 +15,30 @@ using FunctionBase58Encode = FunctionBaseXXConversion<Base58EncodeImpl>;
 }
 REGISTER_FUNCTION(Base58Encode)
 {
-    factory.registerFunction<FunctionBase58Encode>();
+    FunctionDocumentation::Description description = R"(
+Encodes a string using [Base58](https://tools.ietf.org/id/draft-msporny-base58-01.html) encoding.
+)";
+    FunctionDocumentation::Syntax syntax = "base58Encode(plaintext)";
+    FunctionDocumentation::Arguments arguments = {
+        {"plaintext", "Plaintext to encode.", {"String"}}
+    };
+    FunctionDocumentation::ReturnedValue returned_value = {"Returns a string containing the encoded value of the argument.", {"String"}};
+    FunctionDocumentation::Examples examples = {
+    {
+        "Usage example",
+        "SELECT base58Encode('ClickHouse');",
+        R"(
+┌─base58Encode('ClickHouse')─┐
+│ 4nhk8K7GHXf6zx             │
+└────────────────────────────┘
+        )"
+    }
+    };
+    FunctionDocumentation::IntroducedIn introduced_in = {18, 16};
```

Comment:
Sure that 18.16 is the right version? That seems very very early for a rather obscure function.